proposals/0004-FaultTolerance.md

```diff
@@ -0,0 +1,319 @@
+     *
+     * @return The max number of retries. -1 indicates retry forever.
+     * IllegalArgumentException if maxRetries <-1.
+     */
+    int maxRetries() default 3;
+
+    /**
+     * The delay between retries. Defaults to {@link Duration#NONE}.
+     * @return
+     */
+    int delay() default 0;
+
+    /**
+     *
+     * @return the delay unit
+     */
+
+    TimeUnit delayUnit() default TimeUnit.MILLISECONDS;
+
+    /**
+     * @return the maximum duration to perform retries for.
+     */
+    int maxDuration() default 20;
+
+    /**
+     *
+     * @return the duration unit
+     */
+    TimeUnit durationUnit() default TimeUnit.MILLISECONDS;
+
+    /**
+     *
+     * @return the jitter that randomly vary retry delays by. e.g. a jitter of 200 milliseconds
+     * will randomly add betweem -200 and 200 milliseconds to each retry delay.
+     */
+    int jitter() default 200;
+
+    /**
+     *
+     * @return the jitter delay unit.
+     */
+    TimeUnit jitterDelayUnit() default TimeUnit.MILLISECONDS;
+
+    /**
+     * For each retry delay, a randomly portion of the delay multiplied by the jitterFactor will be added or subtracted to the delay.
+     * e.g. a retry delay of 200 milliseconds and a jitter of 0.25 will result in a random retry delay between 150 and 250 milliseconds.
+     * @return the jitter factor.
+     */
+
+    double jitterFactor() default 0.25;
+
+    /**
+     *
+     * @return Specify the failure to retry on
+     */
+    Class<? extends Throwable>[] retryOn() default { Throwable.class };
+
+    /**
+     *
+     * @return Specify the failure to abort on
+     */
+    Class<? extends Throwable>[] aboartOn() default { Throwable.class };
+
+
+}
+
+
+```
+#### CircuitBreaker: a rule to define when to close the circuit
+
+```
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface CircuitBreaker {
+
+        /**
+     * Define the failure criteria
+     * @return the failure exception
+     */
+    Class<? extends Throwable>[] failOn() default Throwable.class;
+
+    /**
+     *
+     * @return The delay time after the circuit is open
+     */
+    long delay() default 2;
+
+    /**
+     *
+     * @return The delay unit after the circuit is open
+     */
+
+    TimeUnit delayUnit() default TimeUnit.SECONDS;
+
+    /**
+     *
+     * @return The failure threshold to open the circuit
+     */
+    long failThreshold() default 2;
+
+    /**
+     *
+     * @return The success threshold to fully close the circuit
+     */
+    long successThreshold() default 2;
+
+}
+```
+#### Fallback
+```
+/**
+ * Define the Fallback annotation to specify the fallback callable, BiConsumer or BiFuncation
+ *@author Emily Jiang
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface Fallback {
+
+    /**
+     *
+     * @return the fallback class
+     */
+    Class<?> fallback() default Callable.class;
```

Comment:
Emily, Antoine and I spoke about this briefly this week, and I think we agreed that the intent was confused to Emily.  Fallback is intended to be a class level annotation only, for now, and the class is meant to share some similarity to the class being executed.  `Callable` isn't valid here.